ssl/ssl_lib.c

```diff
@@ -6469,14 +6576,24 @@ int SSL_alloc_buffers(SSL *ssl)
 
 void SSL_CTX_set_keylog_callback(SSL_CTX *ctx, SSL_CTX_keylog_cb_func cb)
 {
+#ifndef OPENSSL_NO_SSLKEYLOG_CB
     ctx->keylog_callback = cb;
+#else
+     ERR_raise_data(ERR_LIB_SSL, ERR_R_INTERNAL_ERROR,
+                   "Keylogging not supported");
+#endif
 }
 
 SSL_CTX_keylog_cb_func SSL_CTX_get_keylog_callback(const SSL_CTX *ctx)
 {
+#ifndef OPENSSL_NO_SSLKEYLOG_CB
     return ctx->keylog_callback;
+#else
+    return NULL;
+#endif
```

Comment:
I think the logic you have in the setter is enough to neuter the keylog callback entirely. You just don't ever set it, so you can leave it hanging as `NULL` off the `SSL_CTX`.